lib/ansible/modules/unarchive.py

```diff
@@ -907,8 +907,8 @@ def can_handle_archive(self):
         try:
             if self.files_in_archive:
                 return True, None
-        except UnarchiveError:
-            return False, 'Command "%s" could not handle archive.' % self.cmd_path
+        except UnarchiveError as e:
+            return False, 'Command "%s" could not handle archive: %s' % (self.cmd_path, e)
```

Comment:
To keep string compat across python versions we usually wrap exceptions in to_native:
```suggestion
            return False, 'Command "%s" could not handle archive: %s' % (self.cmd_path, to_native(e))
```